crates/wasi-common/yanix/src/fcntl.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{
+    file::{FdFlag, OFlag},
+    Errno, Result,
+};
+use std::os::unix::prelude::*;
+
+pub fn dup_fd(fd: RawFd, close_on_exec: bool) -> Result<RawFd> {
+    // Both fcntl commands expect a RawFd arg which will specify
+    // the minimum duplicated RawFd number. In our case, I don't
+    // think we have to worry about this that much, so passing in
+    // the RawFd descriptor we want duplicated
+    Errno::from_result(unsafe {
+        if close_on_exec {
+            libc::fcntl(fd, libc::F_DUPFD_CLOEXEC, fd)
+        } else {
+            libc::fcntl(fd, libc::F_DUPFD, fd)
+        }
+    })
+}
+
+pub fn get_fd(fd: RawFd) -> Result<FdFlag> {
```

Comment:
Would it be ok to rename this to `get_fd_flags`, which would make it a little easier to read? (and similar for `set_fd`)?

Reply:
Absolutely, and done!